test/test_context.py

```diff
@@ -82,7 +82,7 @@ async def test_should_accept_noncallable_context(self, context):
         assert pipeline.modules[0].context == context
         assert pipeline.modules[0].context == pipeline.modules[1].context
 
-        results = await pipeline.run()
+        results = (await pipeline.run())[0]
```

Comment:
I think this is more readable:
```suggestion
        results, error = await pipeline.run()
```
or (if you don't care about `error`):
```suggestion
        results, _ = await pipeline.run()
```

Replace in other files, too.

Reply:
Indeed, way better